apps/src/lab2/projects/SourcesStore.ts

```diff
@@ -84,7 +84,11 @@ export class RemoteSourcesStore implements SourcesStore {
         replace: replaceExistingVersion,
         firstSaveTimestamp: encodeURIComponent(this.firstSaveTime || ''),
         tabId: getTabId(),
-        projectType: projectType,
+        projectType,
```

Comment:
Is there a reason you found for why we only currently set these options if a `currentVersionId` is available?

Reply:
Tagging in @molly-moen who recently fixed up some of this logic to check me on this - I _think_ the idea is that if there is no current version ID (i.e. this is the first save), then none of the options pertaining to currentVersion/ replace/save time apply which is why we skip creating options at all. However the bug here is that now, not all fields of `SourceUpdateOptions` just pertain to updates/versions, namely `projectType`. I think I'd suggest either:

- Make `currentVersion` optional on `SourceUpdateOptions`, and remove the check for `if (this.currentVersionId)`. If no current version, `replace` should be false, and `firstSaveTimestamp` should be an empty string (based on what it defaults to currently). Always pass through `projectType`
- Split off `projectType` into a separate interface or just pass it through to `sourcesApi.update` as a separate parameter

Let me know I'm missing something though!

Reply:
That seems right to me!

Reply:
Thanks everyone! I went ahead and updated using first option - lmk what you think.

Reply:
So we don't care about any of the other options (other than `projectType`) on initial save? I think I'd be pro Sanchit's second suggestion, ie it might be clearer if we had more explicit types, something like:

```
interface NewSourceOptions {projectType?: string}

interface UpdateSourceOptions extends NewSourceOptions {other options}
```

Reply:
Good point! That makes sense to me - will update.